pkg/diagnostics/types/error.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+)
+
+// DiagnosticError is an error created by the diagnostic framework and has a little
+// more info than a regular error to make them easier to identify in the receiver.
+type DiagnosticError struct {
```

Comment:
It implements the error interface and is passed back as an error return value... it seems a little odd to call DiagnosticMessage{}.Error() but I'll change it if you still think it's clearer.